### ISSUE TITLE
Fix vulnerabilities in the V3 images

### ISSUE THREAD

**Issue by piizei** [Azure/azure-functions-docker#327]
Hi,
Since Alpine image is not going to happen to V3, would be great if the V3 images would fix the CVEs coming from the Debian base image.

CVE-2019-18276 | High | bash
CVE-2019-9923 | High | tar
CVE-2018-6829 | High | libgcrypt20
CVE-2016-2781 | Medium | coreutils
CVE-2019-13627 | Medium | libgcrypt20
CVE-2019-12904 | Medium | libgcrypt20
CVE-2020-3810 | Medium | apt
CVE-2018-1000654 | Medium | libtasn1-6
DSA-4685-1 | Medium | apt
CVE-2019-1551 | Medium | openssl
CVE-2017-18018 | Medium | coreutils
CVE-2011-3374 | Low | apt

**Comment by petenorth** [comment]
Is Microsoft going to respond to this?

**Comment by tomkerkhove** [comment]
Adding @jeffhollan @ahmelsayed

**Comment by balag0** [comment]
Most of these have been addressed now. Only issues reamining should be those with no patch released by debian.

**Comment by petenorth** [comment]
I used GitHub to scan the latest underlying image (which is `mcr.microsoft.com/dotnet/runtime-deps`)

https://github.com/petenorth/dotnet-scan/runs/1426195236?check_suite_focus=true

Zero High and Critical vulnerabilities are normally what constitute the a pass.

**Comment by petenorth** [comment]
I see that `mcr.microsoft.com/dotnet/runtime-deps:3.1-bionic` as a base image would mean no critical or high vulnerabilities.

Is there any reason why a `mcr.microsoft.com/azure-functions/dotnet image` is not available using this base image?

**Comment by balag0** [comment]
Not sure what the differences are, I used the Qualys scan on mcr.microsoft.com/azure-functions/dotnet:3.0
which reports 1 vulnerability which has no patch available as I mentioned above.

![image](https://user-images.githubusercontent.com/28721598/99718550-e3615a80-2a5f-11eb-9823-50ff9cbb3567.png)

the bionic image is an arm specific image, so doesnt work with amd64 image.

The image update policy for the runtime-deps image is at https://hub.docker.com/_/microsoft-dotnet-runtime-deps/ and not sure what tool they use to detect vulnerabilities. But we rely on them to patch OS level vulnerabilties.

**Comment by petenorth** [comment]
There are amd64, arm64 and arm32 architecture bionic runtime dependencies images - it seems odd that only the arm32 architecture image is used to create an Azure Functions 3.0-bionic-arm32v7 why not use the amd64 architecture  bionic runtime dependencies image to create a 3.0-bionic image (with amd64 architecture).

**Comment by balag0** [comment]
I think this is a different question from what we started with :) 
Are we good to close this issue. 

I recommend opening a new issue for this.

**Comment by petenorth** [comment]
Yes I guess so - thanks for the responses.

**Comment by petenorth** [comment]
@balag0 have raised #355

**Comment by ThorstenHans** [comment]
@balag0

Although this is issue and the discussion is quite dated, I want to re-open this one. 

With Azure Defender (Qualys) a first class container image scanning offering (sold by Microsoft) keeps on flagging the [2019-18276 CVE](https://security-tracker.debian.org/tracker/CVE-2019-18276). Obviously, it is caused by downstream Debian (Buster) which is used as base image for `mcr.microsoft.com/azure-functions/dotnet:3.0`.

However, I stumbled upon this issue (https://github.com/dotnet/dotnet-docker/issues/3134) mentioning that official `dotnet` images (including SDK 3.1 which is required IIRC) are now available with Debian bullseye as base.

This would fix the CVE mentioned above and we could finally stop building our own base-images for containerized function workloads.

Long story short, are you planning to move `mcr.microsoft.com/azure-functions/dotnet:3.0` from buster to bullseye in the near future?

**Comment by tomkerkhove** [comment]
@ahmelsayed can you help Thorsten?

**Comment by balag0** [comment]
@ThorstenHans Lets move this to a separate issue if we need further followup.
Short answer is - No plans to move dotnet3 images from buster to bullseye. Is moving to functions v4-dotnet6 an option for you since those images are based on bullseye.

**Comment by ThorstenHans** [comment]
> @ThorstenHans Lets move this to a separate issue if we need further followup. Short answer is - No plans to move dotnet3 images from buster to bullseye. Is moving to functions v4-dotnet6 an option for you since those images are based on bullseye.

@balag0 that's something I've to discuss with the team. From hosting perspective, we've to postpone it till v4 is in GA (IIRC 22nd of November). However, the team has a bunch of code that must be compiled and tested against .NET 6. 

But thanks for the heads-up. I'll file a new issue depending on the outcome of the discussion with the team.
